model/event/AbstractBatteryEvent.php

```diff
@@ -16,40 +16,56 @@
  *
  * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
  */
-
 namespace oat\taoBattery\model\event;
 
-use oat\oatbox\event\Event;
+use core_kernel_classes_Resource as Resource;
 
-class AbstractBatteryEvent implements Event
+/**
+ * Class AbstractBatteryEvent
+ * @package oat\taoBattery\model\event
+ */
+class AbstractBatteryEvent implements BatteryEventInterface
 {
     /**
-     * @var string
+     * @var Resource
+     */
+    private $battery;
+    /**
+     * @var array
      */
-    protected $batteryId;
+    private $newValues;
 
     /**
      * AbstractBatteryEvent constructor.
-     * @param string $batteryId
+     * @param Resource $battery
+     * @param array $newValues
      */
-    public function __construct($batteryId)
+    public function __construct(Resource $battery, array $newValues)
     {
-        $this->batteryId = $batteryId;
+        $this->battery = $battery;
+        $this->newValues = $newValues;
     }
 
     /**
      * @return string
      */
-    public function getBatteryId()
```

Comment:
Just to remind, in `taoNccerDelivery/model/Assignment/RdsExaminationService.php` `getBatteryId()` method is used (in TAO-7230_assignments_rework branch)

Reply:
This is used only for `BatteryBeforeRemoveEvent` class. I will move those methods to this event only. Is it ok?

Reply:
ok (to keep code working after merging develop to TAO-7230_assignments_rework), but mark it as deprecated and later we will remove it and will change a call in taoNccerDelivery/model/Assignment/RdsExaminationService.php

Reply:
Whole class or only method getBatteryId?